src/inmanta/protocol/methods_v2.py

```diff
@@ -1328,3 +1328,52 @@ def get_environment_metrics(
     :param end_interval: The end of the time window for which the metrics should be returned.
     :param nb_datapoints: The amount of datapoint that will be returned within the given time interval for each metric.
     """
+
+
+@typedmethod(path="/login", operation="POST", client_types=[ClientType.api], enforce_auth=False, api_version=2)
+def login(username: str, password: str) -> ReturnValue[model.LoginReturn]:
+    """Login a new user. When the login succeeds an authentication header is returned with the Bearer token set.
+
+    :param username: The user to login
+    :param password: The password of this user
+    :raises UnauthorizedException: Raised when the login failed or if server authentication is not enabled
+    """
+
+
+@typedmethod(path="/user", operation="GET", client_types=[ClientType.api], api_version=2)
+def list_users() -> List[model.User]:
+    """List all users
+
+    :return: A list of all users"""
+
+
+@typedmethod(path="/user/<username>", operation="DELETE", client_types=[ClientType.api], api_version=2)
```

Comment:
Shouldn't this require authentication? Same question for the methods below.

Reply:
I think they all do by default?
only the` /login ` has `enforce_auth=False`

Reply:
Oh, I had missed that it defaulted to `True`. Makes sense though.

Reply:
But how does that work for agent methods? I don't see them being excluded, do they have their own form of auth or is that an oversight?

Reply:
I think that the agents need a token to: 

- the `create_token` method uses the `client_type` to create a token and one of the possible types is agent. 

- then `authorize_request` function verifies that the correct client_type is used.

Reply:
What I mean to say is that enabling authentication also enables it for the agents. I have no clue how difficult it is to set up for the agent, and whether that is acceptable or not.

Reply:
I guess it makes sense.